lib/ansible/module_utils/facts/hardware/linux.py

```diff
@@ -700,7 +700,13 @@ def get_device_facts(self):
 
             sg_inq = self.module.get_bin_path('sg_inq')
 
-            if sg_inq:
+            # we can get NVMe device's serial number from /sys/block/<name>/device/serial
+            serial_path = "/sys/block/%s/device/serial" % (block)
+            if os.path.exists(serial_path):
+                with open(serial_path, 'r') as f:
+                    serial = f.read()
+                    d['serial'] = serial.strip('\n')
+            elif sg_inq:
```

Comment:
Can you flip this logic so that the current behavior is the default, and we fallback to the new behavior?

Reply:
Sure, I can. But I think in this case we are trying to use `sg_inq` when we have simpler way of getting the NVMe device's serial number. Does it sound reasonable?

Reply:
@relrod I've changed it as you suggested to have an old behavior in the first place.

Reply:
> Does it sound reasonable?

The main thing is that we don't want to risk changing the value that people are currently relying on. By keeping the current behavior if it works, we eliminate that and just add a fallback to try if it doesn't.